website/docs/docs/cloud/connect-data-platform/connect-snowflake.md

```diff
@@ -68,3 +72,18 @@ more information on configuring a Snowflake OAuth connection in dbt Cloud, pleas
 ## Configuration
 
 To learn how to optimize performance with data platform-specific configurations in dbt Cloud, refer to [Snowflake-specific configuration](/reference/resource-configs/snowflake-configs).
+
+## Troubleshooting
+<!--might need to turn this into details toggle if more troubleshooting items arise -->
+
+If you're receiving a `Could not deserialize key data` or `JWT token` error, refer to the following causes and solutions:
+
+- **Error: `Could not deserialize key data`**
+  - Possible causes &mdash; This might be due incorrect copying or the not including dashes or commented lines. 
+  - Solution &mdash; You can either manually type the key values or copy the key from its source and paste it into a text editor to verify it before using it in dbt Cloud. 
+
+- **Error: `JWT token`**
+  - Possible causes &mdash; When connecting to Snowflake, dbt gets a JWT token valid for only 60 seconds. If there's no response from Snowflake within this time, you might see a `JWT token is invalid` error in dbt Cloud. Snowflake's 60-second token rule can sometimes reveal issues like NTP server synchronization problems (server clocks not aligned), network latency, or token problems.
+  - Solution &mdash; Confirm mismatches between Snowflake's expected public or private keys or any Snowflake connection parameter errors (such as incorrect login details). Additionally, you can reach out to Snowflake for help or refer to this Snowflake doc for more info: [Key-Based Authentication Failed with JWT token is invalid Error](https://community.snowflake.com/s/article/Key-Based-Authentication-Failed-with-JWT-token-is-invalid-Error). 
+
```

Comment:
(It didn't take in account my comment from here)

We are talking about 2 different things here, not just one:
1. it could be transient issue between Snowflake and dbt Cloud. In that case nothing to do from the user. This would require dbt being able to retry connections
2. it could be due to the fact that the public key is not entered properly on the Snowflake side. In that case, users could check that it is set correctly.

Reply:
Gotcha! Ok I was reading a zendesk ticket on this but will clarify @b-per ✨